mempool/priority/mempool_test.go

```diff
@@ -829,3 +829,90 @@ func TestConcurrentCheckTxDataRace(t *testing.T) {
 	// Wait for the test goroutine to complete
 	wg.Wait()
 }
+
+func TestTxMempool_IsRejectedTx(t *testing.T) {
+	t.Run("CheckTx rejection", func(t *testing.T) {
+		txmp := setup(t, 100)
+
+		// Create a transaction that will be rejected by CheckTx (invalid format)
+		rejectedTx := []byte("invalid-tx-format")
+		txKey := types.Tx(rejectedTx).Key()
+
+		// The transaction should not be rejected initially
+		require.False(t, txmp.IsRejectedTx(txKey))
+
+		// Try to add the transaction - it should be rejected
+		err := txmp.CheckTx(rejectedTx, nil, mempool.TxInfo{})
+		require.NoError(t, err) // CheckTx method itself doesn't return error for app rejection
```

Comment:
why does this not expect an error but then other tests in this file do expect an error?

Reply:
Yeah there's a little bit of an incongruence that we might want to fix. The priority mempool doesn't return an error if the transaction failed in CheckTx. As nina said, it will just have a non zero code and the error message will be in the raw log. In the CAT pool, it wraps the error code and raw log as an error and returns an error